src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -173,3 +179,60 @@ g.test('readwrite_upon_aspects')
       encoder.finish();
     }, !success);
   });
+
+g.test('shader_stages_and_visibility')
+  .params(
+    params()
+      .combine(poptions('readVisibility', kShaderStageCombinations))
```

Comment:
optional nit: Perhaps no need to test all 8x4 combinations, and just [0, kShaderStages] would be fine here too.

Reply:
Agreed. Done!